tests/testthat/test-workflow-bbi.R

```diff
@@ -205,7 +205,47 @@ withr::with_options(list(
       wait_for_nonmem(mod_fail, 2, interval = 1),
         "Expiration was reached"
     )
+  })
+
+  test_that("check_run_times() works with one model [BBR-CRT-001]", {
+    run_times <- check_run_times(mod1, .wait = FALSE)
+    expect_output(str(run_times), "tibble [1 × 3]", fixed = TRUE)
+    expect_output(str(run_times), " $ model_run", fixed = TRUE)
+    expect_output(str(run_times), " $ threads", fixed = TRUE)
+    expect_output(str(run_times), " $ estimation_time", fixed = TRUE)
+  })
+
+  test_that("check_run_times() works with multiple models [BBR-CRT-002]", {
+    run_times <- check_run_times(list(mod1, mod2, mod3), .wait = FALSE)
+    expect_output(str(run_times), "tibble [3 × 3]", fixed = TRUE)
+    expect_output(str(run_times), " $ model_run", fixed = TRUE)
+    expect_output(str(run_times), " $ threads", fixed = TRUE)
+    expect_output(str(run_times), " $ estimation_time", fixed = TRUE)
+  })
+
+  test_that("check_run_times() .return_times arg [BBR-CRT-003]", {
+    run_times <- check_run_times(mod1, .wait = FALSE, .return_times = "all")
+    expect_output(str(run_times), "tibble [1 × 5]", fixed = TRUE)
+    expect_output(str(run_times), " $ model_run", fixed = TRUE)
+    expect_output(str(run_times), " $ threads", fixed = TRUE)
+    expect_output(str(run_times), " $ estimation_time", fixed = TRUE)
+    expect_output(str(run_times), " $ covariance_time", fixed = TRUE)
+    expect_output(str(run_times), " $ cpu_time", fixed = TRUE)
+
+    run_times <- check_run_times(list(mod1, mod2), .wait = FALSE,
+                                 .return_times = c("estimation_time", "covariance_time"))
+    expect_output(str(run_times), "tibble [2 × 4]", fixed = TRUE)
+    expect_output(str(run_times), " $ model_run", fixed = TRUE)
+    expect_output(str(run_times), " $ threads", fixed = TRUE)
+    expect_output(str(run_times), " $ estimation_time", fixed = TRUE)
+    expect_output(str(run_times), " $ covariance_time", fixed = TRUE)
+  })
```

Comment:
@seth127 let me know what you think about this style of testing. There are perhaps better ways to test this, but this seemed the most comprehensive (though sort of looks dumb? - maybe just me)

Reply:
I don't have a strong reason for this, but I would prefer doing something like:

```r
expected_cols <- c("model_run", "threads", "estimation_time", "covariance_time", "cpu_time")

expect_true(all(res_cols %in% names(run_times)))
expect_equal(dim(run_times), c(1, 5))

...
```

Like I said, I don't have a really strong reason to prefer that style, but relying on `str()` printing and doing string matching on it just feels kind of awkward and brittle.